File: src/main/java/fr/eletutour/eweather/dto/Forecast.java
package fr.eletutour.eweather.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

/**
 * @author ewanletutour
 * @version 1.0.0
 * @since 1.0.0
 * <p>
 * Forecast class
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Forecast {
    private double latitude;

    private double longitude;

    private String timezone;

    private String location;

    /**
     * A data point containing the current weather.css conditions at the requested location.
     */
    private Currently currently;

    private Hourly hourly;

    /**
     * A data block containing the weather.css conditions day-by-day for the next week.
     */
    private Daily daily;

    private List<Alert> alerts = new ArrayList<>();
}

File: client/src/app/models/forecast.ts
import {Currently} from './currently';
import {Hourly} from './hourly';
import {Daily} from './daily';

export class Forecast {
  public latitude: number;
  public longitude: number;
  public timezone: string;
  public location: string;
  public currently: Currently;
  public hourly: Hourly;
  public daily: Daily;

  constructor();

  constructor(obj?: any) {
    if (obj) {
      this.latitude = obj.latitude || 0;
      this.longitude = obj.longitude || 0;
      this.timezone = obj.timezone || '';
      this.location = obj.location || '';
      this.currently = obj.currently;
      this.hourly = obj.hourly;
      this.daily = obj.daily;
    } else {
      this.latitude = 0;
      this.longitude = 0;
      this.timezone = '';
      this.location = '';
      this.currently = new Currently();
      this.hourly = new Hourly();
      this.daily = new Daily();
    }

  }
}
